Various updates to fix CVEs

dfarrell07 opened the pull request:
See commit messages for details.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

tpantelis commented on tools/go.mod:
```diff
@@ -1,6 +1,8 @@
-go 1.20
+go 1.21
+
+toolchain go1.21.3
```
What is this? Haven't seen it before...

dfarrell07 replied:
Yeah I'm not sure, I see https://go.dev/ref/mod#go-mod-file-toolchain

tpantelis commented on tools/go.mod:
```diff
@@ -1,6 +1,8 @@
 module github.com/submariner-io/shipyard/tools
 
-go 1.20
+go 1.21
```
Why was this bumped? Note that we specify 1.20 elsewhere.

dfarrell07 replied:
It seems like a dependency requires it. https://stackoverflow.com/questions/77079565/go-mod-tidy-will-auto-upgrade-go-version-in-go-mod

tpantelis replied:
Why wasn't this done/needed in devel? Or these CVE's haven't been applied there yet?

dfarrell07 replied:
Weird. If I use golang `go1.20.10` this doesn't happen. That seems to show the StackOverflow answer is wrong...

I'll try the set again with that go version.

dfarrell07 committed:
Update grpc to fix CVEs

Update generated by `go get -u google.golang.org/grpc`/`go mod tidy`.

Fixes GHSA-m425-mq94-257g and GHSA-qppj-fm5r-hxr3.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Update docker to fix GHSA-jq35-85cj-fj4p

Update generated by `go get -u github.com/docker/docker`/`go mod tidy`.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Update opentelemetry to fix GHSA-rcjv-mgp8-qvmr

Update generated by

```
go get -u go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp
go get -u go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp
go get -u \
  go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
go get -u go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc
go mod tidy
```

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Update otelhttptrace to fix GHSA-rcjv-mgp8-qvmr

```
go get -u \
  go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace
go mod tidy
```

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
The commit message linter keeps globing this line together.

```
8: B1 Line exceeds max length (85>80): "go get -u go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
```

Continuing to try less-good ways.

dfarrell07 commented:
Even written like this, it fails. :exploding_head: 

```
Update opentelemetry to fix GHSA-rcjv-mgp8-qvmr

Update generated by

go get -u

go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace

go mod tidy
```

dfarrell07 commented:
[Clean up clusters](https://github.com/submariner-io/shipyard/actions/runs/6787139241/job/18449547246?pr=1465#logs) failed. I don't have an option to re-run it (yet?).

dfarrell07 commented:
> [Clean up clusters](https://github.com/submariner-io/shipyard/actions/runs/6787139241/job/18449547246?pr=1465#logs) failed. I don't have an option to re-run it (yet?).

I was able to re-run once the rest of the workflow completed. Waiting on results..

dfarrell07 commented:
> > [Clean up clusters](https://github.com/submariner-io/shipyard/actions/runs/6787139241/job/18449547246?pr=1465#logs) failed. I don't have an option to re-run it (yet?).
> 
> I was able to re-run once the rest of the workflow completed. Waiting on results..

And it passed :heavy_check_mark:
